### ISSUE TITLE
mysql.chart.py should not pass config variables to MySQLdb.connect if they are provided via my.cnf

### ISSUE THREAD

**Issue by bbbenjie** [firehol/netdata#972]
Hi guys,
my my.cnf sets a username, password and a socket for MySQL connections, so I may only need to setup this config at mysql.conf:

```
mycnf:
  name     : 'local'
  'my.cnf' : '/path/to/my.cnf'
```

Unfortunately mysql.chart.py sets the username to 'root' and empties the password and socket before it tries to connect to the server, because I only add my.cnf to mysql.conf.

```
if 'user' not in configuration:
    self.configuration['user'] = 'root'
if 'pass' not in configuration:
    self.configuration['pass'] = ''
if 'my.cnf' in configuration:
    self.configuration['socket'] = ''
    self.configuration['host'] = ''
    self.configuration['port'] = 0

[...]

self.connection = MySQLdb.connect(user=self.configuration['user'],
                                  passwd=self.configuration['pass'],
                                  read_default_file=self.configuration['my.cnf'],
                                  unix_socket=self.configuration['socket'],
                                  host=self.configuration['host'],
                                  port=self.configuration['port'],
                                  connect_timeout=self.update_every)
```
This is causing the following error:
```
16-09-17 13:08:42: python.d ERROR: mysql_mycnf Cannot establish connection to MySQL.
16-09-17 13:08:42: python.d DEBUG: mysql_mycnf (2002, "Can't connect to local MySQL server through socket '' (111)")
16-09-17 13:08:42: python.d ERROR: mysql_mycnf check function failed.
16-09-17 13:08:42: python.d INFO: Disabled mysql/mycnf
```
For my needs MySQLdb.connect would work if only read_default_file and connection_timeout is set:
```
self.connection = MySQLdb.connect(read_default_file=self.configuration['my.cnf'],
                                  connect_timeout=self.update_every)
```

**Comment by ktsaou** [comment]
@paulfantom please check this.

**Comment by paulfantom** [comment]
@bbbenjie could you test if changing line 323 in `mysql.chart.py` from:
```py
            self.configuration['user'] = 'root'
```
to:
```py
            self.configuration['user'] = ''
```
solves this issue?

**Comment by bbbenjie** [comment]
Sorry @paulfantom, changing 'root' to '' did not solve this issue.
python.d/MySQLdb still complains a empty socket:
```
16-09-17 18:17:21: python.d INFO: using MySQLdb
16-09-17 18:17:21: python.d DEBUG: mysql: loading module configuration: '/path/to/netdata/etc/netdata/python.d/mysql.conf'
16-09-17 18:17:21: python.d DEBUG: mysql: reading configuration
16-09-17 18:17:21: python.d DEBUG: mysql/mycnf: job added
16-09-17 18:17:21: python.d DEBUG: all job objects [<Service(mycnf, initial daemon)>]
16-09-17 18:17:21: python.d ERROR: mysql_mycnf Cannot establish connection to MySQL.
16-09-17 18:17:21: python.d DEBUG: mysql_mycnf (2002, "Can't connect to local MySQL server through socket '' (111)")
16-09-17 18:17:21: python.d ERROR: mysql_mycnf check function failed.
16-09-17 18:17:21: python.d INFO: Disabled mysql/mycnf
16-09-17 18:17:21: python.d DEBUG: overridden job names: []
16-09-17 18:17:21: python.d DEBUG: all remaining job objects: []
16-09-17 18:17:21: python.d FATAL: no more jobs
DISABLE
```

**Comment by paulfantom** [comment]
@bbbenjie 
I pushed [some changes](https://github.com/paulfantom/netdata/commit/c2678142030ece5da3b2810046229793024c30ab) to my netdata repository, can you test it?

**Comment by bbbenjie** [comment]
@paulfantom, I can confirm that it works fine for me with my.cnf but also with a socket connection.
TCP fails because you removed the integer casting from `self.configuration['port'] = configuration['port']`.
With integer casting, tcp connections also establishes.

**Comment by paulfantom** [comment]
Oh, forgot about it, fixed in #975. I'm probably doing too many things at once :smile:

It needs to be tested more before accepting. I am currently running #975 on my machines and it works. @ktsaou could you run it on some of yours?
